pallets/liquidity-pools/src/message.rs

```diff
@@ -557,6 +562,19 @@ impl LPEncoding for Message {
 	fn empty() -> Message {
 		Message::Batch(BatchMessages::default())
 	}
+
+	fn get_proof(&self) -> Option<Proof> {
+		match self {
+			Message::MessageProof { hash } => Some(*hash),
+			_ => None,
+		}
+	}
+
+	fn to_proof_message(&self) -> Self {
+		let hash = keccak_256(&LPEncoding::serialize(self));
+
+		Message::MessageProof { hash }
+	}
```

Comment:
I think my naming confusion comes from here. In my mind:

```rust
fn message_hash(&self) -> MessageHash {
    //Independently of the kind of message, I should be able to get a hash of it
    keccak_256(&LPEncoding::serialize(self))
}

fn message_proof(&self) -> Self {
    match self {
        Message::MessageProof(..) => self,
        other_msg => Message::MessageProof(other_msg.message_hash())
    }
}
```

Reply:
Just explaining my thoughts about this ^^

Reply:
> I think my naming confusion comes from here. In my mind:
> 
> ```rust
> // Nothing to do with proofs, just give the hash
> fn message_hash(&self) -> MessageHash {
>     //Independently of the kind of message, I should be able to get a hash of it
>     keccak_256(&LPEncoding::serialize(self))
> }
> 
> // Creates a new message containing the hash of message
> // or, return the itself if it's already a proof, without recomputing the hash
> fn message_proof(&self) -> Self {
>     match self {
>         Message::MessageProof(..) => self,
>         other_msg => Message::MessageProof(other_msg.message_hash())
>     }
> }
> ```

Are you proposing this as an alternative @lemunozm ?  AFAICT, this is something different than the current impl of `get_proof` which extracts the proof hash if it exists. Your method will always return a hash independent of the message variant.